### ISSUE TITLE
Refactor proofs contracts to be reusable across chains

### ISSUE THREAD

**Issue by pauldowman** [ethereum-optimism/optimism#12421]
## Description

Currently, new dispute game contracts must be deployed for each new L2 chain because [they take constructor args](https://github.com/ethereum-optimism/optimism/blob/d470c77689821371d452917632f10011a7cfe618/packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol#L160-L169) with values that vary based on the target L2 chain.

As discussed in [the design doc](https://github.com/ethereum-optimism/design-docs/blob/main/protocol/proofs/dispute-game-creators.md), the l2-specific configuration should be set at initialization time rather than at construction. This will allow multiple chains to point to the same implementations, removing the need to deploy new contracts for every chain.

**Comment by ControlCplusControlV** [comment]
oops, missed the tag add it automatically to close, just meant to link

**Comment by ajsutton** [comment]
Yeah the word `fixes` automatically closes.  You can just mention the PR without that to link (e.g. https://github.com/ethereum-optimism/design-docs/pull/153)

**Comment by BlocksOnAChain** [comment]
@ControlCplusControlV ok to close this one now?

**Comment by ControlCplusControlV** [comment]
Still on pause, Drippy got moved in front of it in priority

**Comment by mds1** [comment]
I believe the AnchorStateRegistry no longer has immutables so going forward we can reuse those implementations, but the dispute game contracts do still need to be refactored to remove immutables so the implementations can be shared between all chains. 

Currently, changing the prestate requires deploying new, full dispute game implementations for each chain. These contracts are large (~24kb, right at the size limit), therefore they cost 3-5M gas to deploy. Say it's 4M gas, this means to update prestates we can only update the prestate for 3 chains in one transaction (30M gas limit / 4M gas per game / 2 games per chain = 3.75). This makes the dispute game contracts the bottleneck for scaling our upgrade processes, as we will have to perform multiple signing ceremonies and multiple transactions for any upgrade that requires new dispute game contracts. Therefore we should bump the priority of this issue

**Comment by BlocksOnAChain** [comment]
@ControlCplusControlV can you share more details on this one, since I see that related PR (https://github.com/ethereum-optimism/design-docs/pull/153) for it is merged?
Any other work remaining for you to close this out?

**Comment by ControlCplusControlV** [comment]
That was just the design doc, but now the actual changes + specs need to go in

**Comment by stevennevins** [comment]
Updates to the design doc based on latest direction:
https://github.com/ethereum-optimism/design-docs/pull/275
